test/src/unit/modules/Project.test.js

```diff
@@ -0,0 +1,150 @@
+            project.should.containSubset(args);
+            const id = project.projectID;
+            const name = project.name;
+            project.directory.should.not.equal(`${name}-${id}`);
+        });
+        it('Correctly sets directory to name-id', () => {
+            const project = new Project({
+                name: 'newdummyproject'
+            }, './someworkspace');
+            project.should.be.an('object');
+            const id = project.projectID;
+            const name = project.name;
+            project.should.have.property('directory');
+            project.directory.should.equal(`${name}-${id}`);
+        });
+    })
+    describe('toJSON', () => {
+        it('Checks that toJSON removes fields which shouldn\'t be written to .info file on disk', () => {
+            const obj = {
+                logStreams: 'logstream',
+                loadInProgress: true,
+                loadConfig: 'someconfig'
+            }
+            const project = new Project({ name: 'dummy'}, '/codewind-workspace/');
+            project.should.be.an('object');
+            project.should.containSubset({ name: 'dummy' });
+            project.should.not.containSubset(obj);
+    
+            project.logStreams = obj.logStreams,
+            project.loadInProgress = obj.loadInProgress;
+            project.loadConfig = obj.loadConfig;
+    
+            project.should.containSubset(obj);
+    
+            const json = project.toJSON();
+            json.should.not.containSubset(obj);
+        });
+    })
+    describe('getPort', () => {
+        it('Checks that the port returned is internal as RUNNING_IN_K8S is false', () => {
+            const project = new Project({ name: 'dummy'}, '/codewind-workspace/');
+            project.should.be.an('object');
+            project.ports = {
+                internalPort: 5000,
+                exposedPort: 10000
+            }
+            const port = project.getPort();
+            port.should.equal(5000);
+        });
+        it('Checks that the port returned is exposed as RUNNING_IN_K8S is true', () => {
+            const project = new Project({ name: 'dummy'}, '/codewind-workspace/');
+            global.codewind = { RUNNING_IN_K8S: true };
+            project.should.be.an('object');
+            project.ports = {
+                internalPort: 5000,
+                exposedPort: 10000
+            }
+            const port = project.getPort();
+            port.should.equal(10000);
+            global.codewind = { RUNNING_IN_K8S: false };
+        });
+    })
+    describe('projectPath', () => {
+        it('Checks that projectPath() is workspace+directory', () => {
+            const project = new Project({ name: 'dummy', directory: 'directory' }, '/codewind-workspace/');
+            const projectPath = project.projectPath();
+            projectPath.should.equal('/codewind-workspace/directory');
+        });
+    })
+    describe('isValidName', () => {
```

Comment:
I think this is now an obsolete function, we moved to an API `/invalidNameChers`.

Reply:
That might not be relevant, but if it's not used we may want to get rid.